### PR TITLE
Feature/add score component

### PR DESCRIPTION
**Pull request by kentis**

### CODE REVIEW

File: scripts/components/Dialog/SceneScores.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    for (const [scoreId, score] of Object.entries(this.props.sceneScores)){
+      if(scoreId !== "title"){
+        items.push(<tr><td class="h5p-td h5p-summary-task-title">{score.title ? score.title : scoreId}</td><td class="h5p-td h5p-summary-score-bar">{score.raw}/{score.max}</td></tr>)
+      }
+    }
+    return (
+      <tbody>
+        <tr><td class="h5p-td h5p-summary-task-title" colSpan={2}>{sceneTitle}</td></tr>
+        {items}
+        {/* <tr><td>Total</td><td>{totalScore}</td></tr> */}
```
Comment (boyum) [inline scripts/components/Dialog/SceneScores.js]: ?
```suggestion
```

File: scripts/components/Main.js [hunk 2]
```diff
@@ -149,6 +152,35 @@ export default class Main extends React.Component {
...
+      const interaction = scene.interactions[i];
+      switch(interaction.action.library) {
+        case "H5P.Summary 1.10":
+          sceneScoreCard[i]={title: interaction.label.labelText, raw: 0, max: 1, scaled: 0};
+          break;
+        case "H5P.SingleChoiceSet 1.11":
+          sceneScoreCard[i]={title: interaction.label.labelText, raw: 0, max: interaction.action.params.choices.length, scaled: 0};
+          break;
+        default:
+          // Noop
```
Comment (boyum) [inline scripts/components/Main.js]: `H5P.libraryFromString` will parse the library name and give the library's machine name. Using the machine name only, we don't have to update the version numbers later:


```suggestion
      const libraryName = H5P.libraryFromString(interaction.action.library).machineName;
      switch(libraryName) {
        case "H5P.Summary":
          sceneScoreCard[i]={title: interaction.label.labelText, raw: 0, max: 1, scaled: 0};
          break;
        case "H5P.SingleChoiceSet":
          sceneScoreCard[i]={title: interaction.label.labelText, raw: 0, max: interaction.action.params.choices.length, scaled: 0};
          break;
        default:
          // Noop
```

File: semantics.json [hunk 3]
```diff
@@ -356,6 +357,13 @@
           "AudioRecorder"
         ]
       },
+      {
```
Comment (boyum) [inline semantics.json:368]: When adding fields to semantics.json, remember to add these fields in the right places in every language file as well. Only `label` and `description` properties are needed in the language files.

File: stories/static/three.min.js [hunk 4]
```diff
@@ -151,7 +151,7 @@ H5P.ThreeJS = (function () {
...
-      a};this.setFrameOfReferenceType=function(a){q=a};this.setSession=function(b){h=b;null!==h&&(h.addEventListener("select",c),h.addEventListener("selectstart",c),h.addEventListener("selectend",c),h.addEventListener("end",d),h.baseLayer=new XRWebGLLayer(h,f,{framebufferScaleFactor:k}),h.requestFrameOfReference(q).then(function(b){m=b;a.setFramebuffer(h.baseLayer.framebuffer);v.setContext(h);v.start()}),l=h.getInputSources(),h.addEventListener("inputsourceschange",function(){l=h.getInputSources();console.log(l);
+      a};this.setFrameOfReferenceType=function(a){q=a};this.setSession=function(b){h=b;null!==h&&(h.addEventListener("select",c),h.addEventListener("selectstart",c),h.addEventListener("selectend",c),h.addEventListener("end",d),h.baseLayer=new XRWebGLLayer(h,f,{framebufferScaleFactor:k}),h.requestFrameOfReference(q).then(function(b){m=b;a.setFramebuffer(h.baseLayer.framebuffer);v.setContext(h);v.start()}),l=h.getInputSources(),h.addEventListener("inputsourceschange",function(){l=h.getInputSources();(l);
```
Comment (boyum) [inline stories/static/three.min.js:154]: ?

```suggestion
      a};this.setFrameOfReferenceType=function(a){q=a};this.setSession=function(b){h=b;null!==h&&(h.addEventListener("select",c),h.addEventListener("selectstart",c),h.addEventListener("selectend",c),h.addEventListener("end",d),h.baseLayer=new XRWebGLLayer(h,f,{framebufferScaleFactor:k}),h.requestFrameOfReference(q).then(function(b){m=b;a.setFramebuffer(h.baseLayer.framebuffer);v.setContext(h);v.start()}),l=h.getInputSources(),h.addEventListener("inputsourceschange",function(){l=h.getInputSources();
```

File: scripts/components/Dialog/SceneScores.js (hunk 1 above)
Comment (boyum) [inline scripts/components/Dialog/SceneScores.js]:
```suggestion
```

File: scripts/components/Dialog/ScoreSummary.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          totalScores.max += score.max;  
+        }
+      }
+    }
+    return totalScores;
+  }
+  
+  render() {
+    const items = []
+    const totalScores = this.getToalScores(this.props.scores);
```
Comment (boyum) [inline scripts/components/Dialog/ScoreSummary.js]:
```suggestion
  getTotalScores(scores){
    const totalScores = {score: 0, max:0};
    for(const scene in this.context.params.scenes){
      const sceneScores = scores[scene];
      for(const [id, score] of Object.entries(sceneScores)){
        if(id !== "title") {
          totalScores.score += score.raw;
          totalScores.max += score.max;  
        }
      }
    }
    return totalScores;
  }
  
  render() {
    const items = []
    const totalScores = this.getTotalScores(this.props.scores);
```

File: scripts/components/Dialog/SceneScores.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+export default class SceneTotalScores extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  
+  getSceneTitle(sceneId, sceneTitle){
+    if(sceneTitle !== undefined && sceneTitle !== ""){
+      return `Scene: ${sceneTitle}`
+    }
+    return `Scene: ${sceneId}`
```
Comment (boyum) [inline scripts/components/Dialog/SceneScores.js]: Should these be translated?

File: scripts/components/Main.js (hunk 2 above)
Comment (boyum) [reply scripts/components/Main.js]: Why was this resolved?

File: scripts/components/Dialog/SceneScores.js (hunk 6 above)
Comment (kentis) [reply scripts/components/Dialog/SceneScores.js]: yes

**Review by boyum — APPROVED** [review]

**Review by boyum — CHANGES_REQUESTED** [review]
Waaait let me just test

**Review by boyum — APPROVED** [review]

**Pull request merged**